praktikum/poglavlja/12_dinamicka_alokacija/index.rst

```diff
@@ -0,0 +1,104 @@
+Dinamička alokacija memorije
+============================
+
+Podeli svoje utiske o poglavlju na anketi https://docs.google.com/forms/d/1xdwvmDBf-dHSrliFcBA9KL1egNNWD9tYxEjdQRzcqSg
+
+Preduslovi za rad:
+
+.. literalinclude:: objasnjenja/preduslovi-za-rad.txt
+   :language: none
+
+Dinamičku alokaciju koristimo kada ne znamo unapred, pre pokretanja programa, koliko nam je memorije potrebno za rad. Do sada smo
+u radu sa nizovima uvek unapred određivali veličinu niza. Kada ovako jednom odredimo veličinu niza, ne možemo je kasnije menjati.
+Ta memorija nam može postati mala u nekom trenutku. Problem nedovoljne memorije rešavamo tako što je zauzimamo u toku izvršavanja
+programa, tj. alociramo je dinamički. Dinamički alociranu memoriju je potrebno uvek nakon korišćenja osloboditi.
+
+Funkcije za dinamičku alokaciju memorije su ``malloc``, ``calloc``, ``realloc`` i ``free``. Sve funkcije se nalaze u biblioteci ``stdlib.h``,
+pa je potrebno uključiti je kako bi ih koristili. Funkcije ``malloc`` i  ``calloc`` služe za zauzimanje memorije, a funkcija ``realloc``
+za proširenje/smanjene prethodno dinamički zauzete memorije. Ove tri funkcije vraćaju pokazivač na zauzetu memoriju ukoliko su je uspešno
+zauzele ili NULL vrednost ukoliko ne mogu da zauzmu memoriju. Imajući ovo u vidu, uvek je potrebno proveriti da li su ove funkcije uspešno
+zauzele memoriju i izaći iz programa ako nisu. Funkcija ``free`` služi za oslobađanje dinamički zauzete memorije.
+
+Free
+----
+Funkcija ``free`` služi za oslobađanje dinamički zauzete memorije. Očekuje pokazivač na dinamički zauzetu memoriju kako bi je oslobodila.
+Osnovna forma funkcije je::
+
+  void free(void *ptr);
+
+Malloc
+------
+
+Funkcija ``malloc`` se koristi za dinamičku alokaciju memorijskog bloka određene veličine. Funkcija vraća generički pokazivač
+tipa ``void *``, koji se može pretvoriti u bilo koji pokazivač. Ne inicijalizuje zauzetu memoriju na početnu vrednost.
+Osnovna forma funkcije je::
+
+  void *malloc(size_t size);
+
+gde size_t size predstavlja veličinu memorije koju će ``malloc`` pokušati da zauzme.
+
+.. literalinclude:: primeri/malloc.c
+   :linenos:
+
+U primeru iznad unosimo prvo dužinu niza, nakon čega zauzimamo memoriju za tačno toliko članova niza korišćenjem funkcije ``malloc``.
+Za veličinu jednog memorijskog bloka koja nam je potrebna koristimo operator ``sizeof``. Ovaj operator očekuje da mu se prosledi tip podatka
+i vraća količinu memorije koju taj tip podatka zauzima u bajtovima. Kako je nama potrebna memorija za niz ``int`` tipa, sa ``sizeof(int)``
+dobićemo memoriju jednog integer-a, dok ćemo sa ``n * sizeof(int)`` dobiti memoriju za ceo niz. Kada smo pokušali da zauzememo memoriju, potrebno je da
+proverimo da li je ona uspešno zauzeta. Ukoliko je ``malloc`` vratio NULL, izlazimo iz programa, ukoliko nije, unosimo članove niza, pa ih ispisujemo.
+Na kraju programa, oslobađamo dinamički zauzetu memoriju sa funkcijom ``free``.
```

Comment:
Rekao bih "kada nam više nije potrebna" ili "kada smo završili sve što smo planirali" umesto "na kraju programa". Za naše primere je tačno da oslobađamo na kraju programa, ali generalno gledano hoćemo da ih naučimo da sa dinamičkom memorijom postupaju tako da je oslobode kad im više ne treba, zbog programa koji rade u realnom vremenu 24/7/365.